src/components/EnvironmentOverride/SecretOverrides.tsx

```diff
@@ -240,6 +247,24 @@ export function OverrideSecretForm({ name, toggleCollapse }) {
             dispatch({ type: 'createErrors' })
             return
         }
+        if (type === 'volume' && isFilePermissionChecked) {
+            if (!state.filePermission.value) {
+                dispatch({ type: 'filePermission', value: { value: state.filePermission.value, error: "Field is mandatory" } })
+                return
+            }
+            if (state.filePermission.value.startsWith("0")) { //Octal Format
```

Comment:
invert the logic, first check length 4 and then start with 0, also this seems like duplicate logic, can we abstract it out?